p2p/netaddress.go

```diff
@@ -168,17 +168,25 @@ func (na *NetAddress) DialString() string {
 	)
 }
 
-// Dial calls net.Dial on the address.
+// Dial calls net.Dial on the address. If no IP is set, it returns
+// IPNotSetError.
 func (na *NetAddress) Dial() (net.Conn, error) {
+	if na.IP == nil {
```

Comment:
What are the scenarios the `Dial` methods are called when the `IP` is `nil`? Isn't that catched in the constructors up top?